cmd/security-agent/subcommands/runtime/command.go

```diff
@@ -735,6 +735,8 @@ func StartRuntimeSecurity(log log.Component, config config.Component, hostname s
 }
 
 func downloadPolicy(log log.Component, config config.Component, downloadPolicyArgs *downloadPolicyCliParams) error {
+	var err error
```

Comment:
I believe this pattern only works if you use a named return val. 

Not working: https://go.dev/play/p/6sjlSq1Rh-P (output is `<nil>`)
Working: https://go.dev/play/p/VvWqYwXu39i (output is `cerr`)

Reply:
i think we'd want to do this pattern from the article you linked ```func doSomething() error {
    f, err := os.Create("foo")
    if err != nil {
        return err
    }
    defer f.Close()

    if _, err := f.Write([]byte("bar"); err != nil {
        return err
    }

    if err := f.Close(); err != nil {
        return err
    }
    return nil
}
```

Reply:
Thank you for catching this ! My new commit should implement this pattern and fix this issue.